### DISCUSSION TITLE
What is the purpose of jax.random.split? Can't I just create new random keys?

### DISCUSSION THREAD

**Discussion by brunomorayib** [jax-ml/jax#13645] [Q&A] [answered]
Instead of doing, e.g. `random.split(random.PRNGKey(0),3)` can't I just create 3 different keys? In what sense the random seeds are "split" ?

**Comment by jakevdp** [comment, marked as answer]
The purpose of `random.split` is to allow you to generate an arbitrary number of independent pseudorandom values given a single key. You can certainly directly create as many random keys as you wish, but requiring this in every case would be problematic.

As an example, consider an iterative solver that creates a new random vector each iteration. If keys had to be constructed manually, the user would have to pass-in an arbitrarily large number of random keys to the solver. With `random.split`, passing a single key is sufficient, and the function can split it as needed to generate as many independent pseudorandom values as needed.

For more background on the design of JAX's random generators, including a section motivating the need for the *split* operation, see [JAX PRGN Design Doc](https://jax.readthedocs.io/en/latest/jep/263-prng.html).

**Reply by brunomorayib** [reply]
Thanks! Understood now :)
